doc/subsystems/networking/vlan.rst

```diff
@@ -0,0 +1,49 @@
+.. _vlan-support:
+
+Virtual LAN (VLAN) Support
+##########################
+
+Overview
+********
+
+`Virtual LAN <https://wikipedia.org/wiki/Virtual_LAN>`_ (VLAN) is a
+partitioned and isolated computer network at the data link layer
+(OSI layer 2). For ethernet network this refers to
+`IEEE 802.1Q <https://en.wikipedia.org/wiki/IEEE_802.1Q>`_
+
+In Zephyr, each individual VLAN is modelled as a virtual network interface.
+This means that there is a ethernet network interface that corresponds to
+real physical ethernet port in the system. A virtual network interface is
+created for each VLAN, and this virtual network interface connects to the
+real network interface. This is similar as how Linux implements VLANs. The
+*eth0* is the real network interface and *vlan0* is a virtual network interface
+which is run on top of *eth0*.
+
+The VLAN support must be enabled at compile time. The option to set is
```

Comment:
Here's a try at clarifying this paragraph:

    VLAN support must be enabled at compile time by
    setting option :option:`CONFIG_NET_VLAN` and also
    setting option :option:`CONFIG_NET_VLAN_COUNT`
    to reflect how many network interfaces there will be
    in the system.  For example, if there is one network
    interface without VLAN support, and two with VLAN support,
    the :option:`CONFIG_NET_VLAN_COUNT` should
    be set to 3.